libs/ui-watt/src/lib/components/button/watt-button.component.ts

```diff
@@ -14,106 +14,49 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import {
-  ChangeDetectionStrategy,
-  Component,
-  HostBinding,
-  Injector,
-  Input,
-  Type,
-  ViewEncapsulation,
-} from '@angular/core';
-
-import { disabledAttributeToken } from './disabled-attribute-token';
-import { typeAttributeToken } from './type-attribute-token';
-import { WattButtonSize } from './watt-button-size';
-import { WattButtonType } from './watt-button-type';
-import { WattButtonVariant } from './watt-button-variant';
-import { WattIcon } from './../../foundations/icon';
-import { WattPrimaryButtonComponent } from './primary-button/watt-primary-button.component';
-import { WattSecondaryButtonComponent } from './secondary-button/watt-secondary-button.component';
-import { WattTextButtonComponent } from './text-button/watt-text-button.component';
+import { Component, Input, ViewEncapsulation } from '@angular/core';
+import { WattIcon } from '../../foundations/icon';
+
+export const WattButtonTypes = [
+  'primary',
+  'secondary',
+  'text',
+  'icon',
+] as const;
+export type WattButtonVariant = typeof WattButtonTypes[number];
+export type WattButtonSize = 'normal' | 'large';
+export type WattButtonType = 'button' | 'reset' | 'submit';
 
 @Component({
-  changeDetection: ChangeDetectionStrategy.OnPush,
-  encapsulation: ViewEncapsulation.None,
   selector: 'watt-button',
+  template: `
+    <button
+      mat-button
+      [ngClass]="['watt-button--' + variant, 'watt-button--' + size]"
+      [disabled]="disabled"
+      type=""
+    >
+      <watt-spinner
+        *ngIf="loading"
+        [diameter]="18"
+        class="content-grid-item content-grid-item-spinner"
+      ></watt-spinner>
+      <watt-icon *ngIf="!loading && hasIcon()" [name]="icon"></watt-icon>
+      <ng-content *ngIf="!loading && variant !== 'icon'"></ng-content>
```

Comment:
issue: the problem with the "if"/"else" approach here is that the button's width will jump when switching between text and spinner. That is when the button (with text) is wider than the `min-width`. 

The "previous" version of the button used a `.content-grid` + `.content-grid-item` approach that position the spinner absolutely and visually hiding the text content. This way the button's width was kept. Consider doing the same.